internal/services/gateway/action/invitation.go

```diff
@@ -0,0 +1,265 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package action
+
+import (
+	"context"
+
+	"agola.io/agola/internal/errors"
+	"agola.io/agola/internal/services/config"
+	"agola.io/agola/internal/services/gateway/common"
+	"agola.io/agola/internal/util"
+	csapitypes "agola.io/agola/services/configstore/api/types"
+	cstypes "agola.io/agola/services/configstore/types"
+	"agola.io/agola/services/gateway/api/types"
+	gwapitypes "agola.io/agola/services/gateway/api/types"
+)
+
+type OrgInvitationResponse struct {
+	Organization  *cstypes.Organization
+	OrgInvitation *cstypes.OrgInvitation
+}
+
+func (h *ActionHandler) GetUserOrgInvitations(ctx context.Context, userRef string, limit int) ([]*OrgInvitationResponse, error) {
+	cOrgInvitations, _, err := h.configstoreClient.GetUserOrgInvitations(ctx, userRef, limit)
+	if err != nil {
+		return nil, util.NewAPIError(util.KindFromRemoteError(err), err)
+	}
+
+	res := make([]*OrgInvitationResponse, len(cOrgInvitations))
+	for i, r := range cOrgInvitations {
+		org, _, err := h.configstoreClient.GetOrg(ctx, r.OrganizationID)
+		if err != nil {
+			return nil, util.NewAPIError(util.KindFromRemoteError(err), err)
+		}
+
+		res[i] = &OrgInvitationResponse{
+			OrgInvitation: r,
+			Organization:  org,
+		}
+	}
+
+	return res, nil
+}
+
+func (h *ActionHandler) GetOrgInvitations(ctx context.Context, orgRef string, limit int) ([]*cstypes.OrgInvitation, error) {
+	if !common.IsUserLogged(ctx) {
+		return nil, errors.Errorf("user not logged in")
+	}
+
+	isOrgOwner, err := h.IsOrgOwner(ctx, orgRef)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to determine ownership")
+	}
+	if !isOrgOwner {
+		return nil, util.NewAPIError(util.ErrForbidden, errors.Errorf("user not authorized"))
+	}
+
+	orgInvitations, _, err := h.configstoreClient.GetOrgInvitations(ctx, orgRef, limit)
+	if err != nil {
+		return nil, util.NewAPIError(util.KindFromRemoteError(err), err)
+	}
+	return orgInvitations, nil
+}
+
+type CreateOrgInvitationRequest struct {
+	UserRef         string
+	OrganizationRef string
+	Role            cstypes.MemberRole
+}
+
+func (h *ActionHandler) CreateOrgInvitation(ctx context.Context, req *CreateOrgInvitationRequest) (*OrgInvitationResponse, error) {
+	if !common.IsUserLogged(ctx) {
+		return nil, errors.Errorf("user not logged in")
+	}
+
+	if h.organizationMemberAddingMode != config.OrganizationMemberAddingModeInvitation {
+		return nil, util.NewAPIError(util.ErrUnauthorized, errors.Errorf("user members can not added by invitation"))
```

Comment:
This cannot be Unauthorized (must only be used when not authenticated). Use BadRequest.